Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/Mandate/Notification.hs

```diff
@@ -0,0 +1,160 @@
+module SharedLogic.Allocator.Jobs.Mandate.Notification (sendPDNNotificationToDriver) where
+
+import qualified Data.Map as M
+import qualified Data.Map.Strict as Map
+import Domain.Types.DriverFee as DF
+import Domain.Types.DriverInformation as DI
+import Domain.Types.DriverPlan as DPlan
+import qualified Domain.Types.Invoice as INV
+import Domain.Types.Mandate (Mandate)
+import Domain.Types.Merchant
+import Domain.Types.Merchant.TransporterConfig
+import qualified Domain.Types.Notification as NTF
+import Domain.Types.Person as P
+import Domain.Types.Plan as Plan
+import qualified Kernel.External.Payment.Interface.Types as PaymentInterface
+import Kernel.Prelude
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Types.Error
+import Kernel.Types.Id (Id, cast)
+import Kernel.Utils.Common
+import qualified Lib.Payment.Domain.Action as APayments
+import Lib.Scheduler
+import Lib.Scheduler.JobStorageType.SchedulerType (createJobIn)
+import SharedLogic.Allocator
+import SharedLogic.DriverFee (changeAutoPayFeesAndInvoicesForDriverFeesToManual)
+import qualified Storage.CachedQueries.Merchant.TransporterConfig as SCT
+import qualified Storage.Queries.DriverFee as QDF
+import qualified Storage.Queries.DriverInformation as QDI
+import qualified Storage.Queries.DriverPlan as QDP
+import qualified Storage.Queries.Invoice as QINV
+import qualified Storage.Queries.Notification as QNTF
+import qualified Tools.Payment as TPayment
+
+sendPDNNotificationToDriver ::
+  ( CacheFlow m r,
+    EsqDBFlow m r,
+    Esq.EsqDBReplicaFlow m r,
+    MonadFlow m,
+    EncFlow m r,
+    HasShortDurationRetryCfg r c,
+    HasField "maxShards" r Int,
+    HasField "schedulerSetName" r Text,
+    HasField "schedulerType" r SchedulerType,
+    HasField "jobInfoMap" r (M.Map Text Bool)
+  ) =>
+  Job 'SendPDNNotificationToDriver ->
+  m ExecutionResult
+sendPDNNotificationToDriver Job {id, jobInfo} = withLogTag ("JobId-" <> id.getId) do
+  ----- added measure duration for debugging -------
```

Comment:
We needed to catch exceptions here also like driverFee scheduler. cc: @aayushagarwal2001